src/PageHeader/manageResponsiveValueWithCSS.ts

```diff
@@ -0,0 +1,130 @@
+      return value ? 'none' : 'flex'
+    })
+ * @returns
+ * {
+ *   "@media screen and (max-width: 768px)": {
+ *     "display": "none"
+ *   },
+ *   "@media screen and (min-width: 768px)": {
+ *     "display": "none"
+ *   },
+ *   "@media screen and (min-width: 1440px)": {
+ *     "display": "flex"
+ *   }
+ * }
+ * 
+ * * @example
+ * CSSManagedResponsiveValue({regular: 'border.default', wide: 'canvas.inset'}, 'backgroundColor', (value): string => {
+    return value
+  })
+ * @returns
+ * {
+ *   "@media screen and (min-width: 768px)": {
+ *     "backgroundColor": "border.default"
+ *   },
+ *   "@media screen and (min-width: 1440px)": {
+ *     "backgroundColor": "canvas.inset"
+ *   }
+ * }
+ * 
+ * * @example
+* CSSManagedResponsiveValue({narrow: 'filled', regular: 'line'}, 'height', (value): string => {
+    return {
+      filled: 8,
+      line: 1,
+    }[value]
+  })
+ * @returns
+ * {
+ *  "@media screen and (max-width: 768px)": {
+ *     "height": 8
+ *   }
+ *   "@media screen and (min-width: 768px)": {
+ *     "height": 1
+ *   },
+ * }
+ */
+export function CSSManagedResponsiveValue<T>(
+  value: T,
+  cssProperty: string,
+  comparisonFunc: (value: T) => void,
+): BetterSystemStyleObject {
+  if (isResponsiveValue(value)) {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const responsiveValue = value as Extract<T, ResponsiveValue<any>>
+
+    // Build media queries with the giving cssProperty and comparisonFunc
+    const narrowMediaQuery =
+      'narrow' in responsiveValue
+        ? {
+            [`@media screen and (max-width: 768px)`]: {
+              [cssProperty]: comparisonFunc(responsiveValue.narrow),
+            },
+          }
+        : {}
+
+    const regularMediaQuery =
+      'regular' in responsiveValue
+        ? {
+            [`@media screen and (min-width: 768px)`]: {
+              [cssProperty]: comparisonFunc(responsiveValue.regular),
+            },
+          }
+        : {}
+
+    const wideMediaQuery =
+      'wide' in responsiveValue
+        ? {
+            [`@media screen and (min-width: 1440px)`]: {
+              [cssProperty]: comparisonFunc(responsiveValue.wide),
+            },
+          }
+        : {}
+
+    // check if all values are the same - this is not a recommended practise but we still should check for it
+    if (areAllValuesTheSame(responsiveValue)) {
+      // if all the values are the same, we can just use one of the value to determine the CSS property's value
+      return {[cssProperty]: comparisonFunc(responsiveValue.narrow)}
+      // check if regular and wide have the same value, if so we can just return the narrow and regular media queries
+    } else if (haveRegularAndWideSameValue(responsiveValue)) {
+      return {
+        ...narrowMediaQuery,
+        ...regularMediaQuery,
+      }
+    } else {
+      return {
+        ...narrowMediaQuery,
+        ...regularMediaQuery,
+        ...wideMediaQuery,
+      }
+    }
```

Comment:
How might we handle fallback/default values in this function?

For example, if a value for the `narrow` viewport is not defined in the responsive value object (`{ regular: true, wide: false }`), how do we specify what value the property should fallback too in the narrow viewport range?

Let me know if that makes sense.

Reply:
It does! And thats for commenting about that. 

I took the fallback value out of the function and made it to be declared with the CSS property within the `sx` prop inside. Josh suggested to that (see the convo [here](https://github.com/primer/react/pull/2662#issuecomment-1350383980)) and it made sense to me because it feels more imperative and an intentional choice than providing a single value.  Let me give you an example to explain it better. 

Let's take this hidden prop as an example
```
hidden={{regular:true, wide: false}}
```
If we provide the fallback value in the function it could be like this
```
CSSManagedResponsiveValue(hidden, 'display', value => {
      return value ? 'none' : 'flex'
}, false)
```
and this will resolve into

```
{
    [@media screen and (max-width: 768px)]: {
       display :'none'  // fallback
    },
    [@media screen and (min-width: 768px)]: {
       display : 'none'
    },
    [@media screen and (max-width: 768px)]: {
       display : 'flex'
    },
}
```

With the current version where we took the fallback value out of the func, now the developer (who uses the func) needs to provide an explicit CSS declaration for that fallback value. (And I am sorry that I forgot to add them 🙈 - maybe missing this led to the confusion or contributed to it. )

function
```
CSSManagedResponsiveValue(hidden, 'display', value => {
      return value ? 'none' : 'flex'
})
```
this will resolve into

```
{
    [@media screen and (min-width: 768px)]: {
       display : 'none'
    },
    [@media screen and (max-width: 768px)]: {
       display : 'flex'
    },
}
```

and in the component itself we will need to add the CSS declaration that uses the fallback value.
```
sx={merge<BetterSystemStyleObject>(
    {
      display: 'none', // `display`: is our CSS property that we are managing responsively and `none` is the fallback value.
      ......
      ...CSSManagedResponsiveValue(hidden, 'display', value => {
        return value ? 'none' : 'flex'
      }),
    },
    sx,
  )}
```

Let me know what you think and if you have any concerns 🙌🏼

Reply:
I like that! Thanks for explaining 👍